src/problem.rs

```diff
@@ -0,0 +1,28 @@
+use nalgebra::{storage::ContiguousStorageMut, ComplexField, Dim, Matrix, Vector};
+
+/// A least squares minimization problem.
+///
+/// This is what [`LevenbergMarquardt`](struct.LevenbergMarquardt.html) needs
+/// to compute the residuals and the Jacobian. See the [module documentation](index.html)
+/// for a usage example.
+pub trait LeastSquaresProblem<F, N, M>
+where
+    F: ComplexField,
+    N: Dim,
+    M: Dim,
+{
+    /// Storage type used for the residuals. Use `nalgebra::storage::Owned<F, M>`
+    /// if you want to use `VectorN` or `MatrixMN`.
+    type ResidualStorage: ContiguousStorageMut<F, M>;
+    type JacobianStorage: ContiguousStorageMut<F, M, N>;
+    type ParameterStorage: ContiguousStorageMut<F, N> + Clone;
+
```

Comment:
As mentioned elsewhere, we should put a `params` into here so that the initial state can just be passed in through the `LeastSquaresProblem`.